### ISSUE TITLE
Height of table incorrectly calculated under zoom

### ISSUE THREAD

**Issue by john-swallow-cenduit** [handsontable/handsontable#7585]
### Description
Height of table container is incorrectly calculated under different zoom levels leading to faulty spacing and in the worst case overlap of subsequent controls. 

Possible Angular specific issue. Only observed under Angular conditions.

Observed under https://jsfiddle.net/api/post/library/pure/ with minimal mods:

- Added non-commercial license key to get rid of the message. If message is present the issue does not show up.
- Added a <button> directly under the table.

### Steps to reproduce
1. Open the demo link and modify the zoom.
2. Observe that the space between the table and the button that follows is variable.
3. Set zoom to 50% and observe that the button appears INSIDE the table.

The height of the container div does NOT align with the content of the table

![image](https://user-images.githubusercontent.com/58601615/108877900-98735800-75f7-11eb-9e87-30f8b8524ec0.png)

### Demo
https://jsfiddle.net/9zmynvfk/

### Your environment
* Handsontable version: 8.3.1. @handsontable/angular 6.0.1
* Browser Name and version: Chrome Version 88.0.4324.182 (Official Build) (64-bit)
* Operating System: Windows 10

**Comment by john-swallow-cenduit** [comment]
Actually reproducible without any change.
Just go to the docs for getting started with HoT and Angular 
https://handsontable.com/docs/8.3.1/frameworks-wrapper-for-angular-simple-example.html
Click the link for "Edit" on the example. 
Then zoom to 50%. The license key warning itself overwrites the grid.

![image](https://user-images.githubusercontent.com/58601615/108885909-44b93c80-7600-11eb-8c11-9995361dbb78.png)

**Comment by AMBudnik** [comment]
Thi @john-swallow-cenduit 

Thank you for sharing. We already have a couple of issues related to zoom in/out on the Github issue board. They are listed here https://github.com/handsontable/handsontable/issues/7527 I think that the first issue on that list describes your case. Please confirm.

**Comment by john-swallow-cenduit** [comment]
I am well aware of #7527 having checked for existing issues before raising this one. However I don't think that this particular issue is the same as the one you reference. The first item on that item #3118 relates to misaligning rows when row headers are used. The above issue occurs regardless of whether row headers are on or off so I cannot see how it adequately covered by that item. 

All the evidence I can see is that this is an Angular specific problem as I can't make it happen in the same way without it - though I don't know why. Though it is related to zoom and as such you might want to deal with under #7525 as far as I can see it doesn't match any of the issues referred to on #7527 except for the fact that is it is an issue when you zoom!

**Comment by john-swallow-cenduit** [comment]
This issue is potentially much more closely related to #6280 which is at least ABOUT the height of the table. #6280 states that the table height is ALWAYS bigger than the space needed. This is not my experience. If your zoom level is > 100% then this seems to be true but if it is < 100% then it is often LESS than the size of the space needed and as such is a much WORSE bug as it means any content after the table starts to appear in the middle and that's just much worse for me. I am also using Angular rather than React as that bug states so I think this bug is a sibling to that one NOT a duplicate of it.

**Comment by AMBudnik** [comment]
You are, @john-swallow-cenduit that issue has to be defined separately and not called a duplicate. 

Interestingly some of those issues relate to a specific numeric value of the zoom. Most of them relate to a situation when there is a 0.3px value that is being rounded in the the result of that rounding we get a rows/header misalignment. 
However, here we have 50% and 33% (I've tested 100-300%, and 90 - 33%) and the table itself does not change much but it overlaps the button underneath.

Let me then add the issue to the https://github.com/handsontable/handsontable/issues/7527 as a separate case. I would also want to add that the fact that I closing the issue does not mean that the investigation is over. We can still exchange ideas and cases to cover the subject as much as needed.

**Comment by adrianszymanski89** [comment]
Hi @john-swallow-cenduit 

We were retesting this issue, and it seems it's no longer present. Can you please confirm that?

**Comment by john-swallow-cenduit** [comment]
TLDR: it's better now but I still think it isn't right.

I would agree that it appears now to be impossible to reproduce the worst symptom, at least under my initial conditions. Under 14.1 I can't produce a situation where my un-styled content pasted in under the table overlays the table. But the behaviour of the table is still a bit odd and would be counter to reasonable expectations.

https://jsfiddle.net/mzheynfu/1/

To make the above I have taken the jsfiddle from https://handsontable.com/docs/javascript-data-grid/angular-basic-example/
And I just pasted in the un-styled button immediately after the HoT.  "<button>My button</button>".  The table has height "auto". 

In an ideal world the behaviour that I would expect would be that the button is flush, or exteremely close to flush, against the bottom of the table in all all zoom situations. However, zooming in and out still produces a result where the spacing between the **visible** bottom of the table varies widely:

1)  Sometimes the button is against the bottom of the table. This is on all zoom levels > 100% that are not fractional. So 200% is flush, 300% is flush.
2) Sometimes there's  a gap. 125%, 150%, etc.
3) Sometimes the table (which is on auto size) brings in a scroll bar and the button is immediately up against the bottom of this scrollable area. This happens for zoom levels < 100%

The scrollbar on zooms < 100% can be observed under a completely unchanged jsdfiddle straight from your own documentation: https://jsfiddle.net/api/post/library/pure/. Reduce the zoom to < 100% and scrollbars will kick in and the full table will not be displayed - which I think is odd for a table set to "auto".

**Comment by adrianszymanski89** [comment]
Hi @john-swallow-cenduit 

Thank you for the update and the new details. I can confirm those issues, and I've updated our internal report about them. As you mentioned, it seems strictly connected to the `height: 'auto` property. It also causes other displaying problems, and we don't recommend using it until we fix it.

**Comment by wuzaofeng** [comment]
This issue can be fixed by modifying the browser's scrollbar style. There may be compatibility issues, but at least it can solve some browsers to avoid line spacing caused by calculation errors

**Comment by adrianszymanski89** [comment]
Hi @wuzaofeng 

Thank you for sharing this solution. It's good to have another workaround before we handle it at the core.
